src/sentry/users/web/accounts.py

```diff
@@ -374,3 +390,79 @@ def confirm_email(request: HttpRequest, user_id: int, hash: str) -> HttpResponse
         )
     messages.add_message(request, level, msg)
     return HttpResponseRedirect(reverse("sentry-account-settings-emails"))
+
+
+@set_referrer_policy("strict-origin-when-cross-origin")
+@login_required
+@control_silo_function
+def confirm_signed_email(
+    request: HttpRequest, signed_data: str
+) -> HttpResponseRedirect | HttpResponse:
+    EMAIL_CONFIRMATION_SALT = options.get("user-settings.signed-url-confirmation-emails-salt")
+
+    use_signed_urls = options.get("user-settings.signed-url-confirmation-emails")
+    if not use_signed_urls:
```

Comment:
is it weird if somebody tries to add a secondary email, then we disable the option, and they can't verify it? should we tell them to add it again here?

Reply:
I did not think of that. It would be a weird experience.

I'll update the error message.